### PR TITLE
Fixing issue where event listeners were not being removed after finalizing DeckGL when used with google maps

### PR DESCRIPTION
**Pull request by puopg**
Was this a regression during the TS refactor?

### CODE REVIEW

**Comment by Pessimistress** [comment]
Thank you for identifying the issue. In 8.8 we moved `userData` to the `Deck` instance for proper type check, but it seems that the google-maps module is not assigning to it.

https://github.com/visgl/deck.gl/blob/574caf1e43a184b3822e1e44d45f3912588592e4/modules/google-maps/src/utils.ts#L47-L50

Should do something like:

https://github.com/visgl/deck.gl/blob/574caf1e43a184b3822e1e44d45f3912588592e4/modules/mapbox/src/deck-utils.ts#L76-L77

**Comment by puopg** [comment]
Ok lemme update this then to abide by the correct format

File: modules/google-maps/src/utils.ts [hunk 1]
```diff
@@ -6,6 +6,16 @@ import type {MjolnirGestureEvent, MjolnirPointerEvent} from 'mjolnir.js';
...
 const MAX_LATITUDE = 85.05113;
 
+type UserData = {
+  _googleMap: google.maps.Map;
+  _eventListeners: {
+    click: google.maps.MapsEventListener;
+    dblclick: google.maps.MapsEventListener;
+    mousemove: google.maps.MapsEventListener;
+    mouseout: google.maps.MapsEventListener;
+  };
```
Comment (Pessimistress) [inline modules/google-maps/src/utils.ts:16]:
```suggestion
  _eventListeners: Record<string, google.maps.MapsEventListener>;
```

**Review by Pessimistress — APPROVED** [review]

**Comment by puopg** [comment]
@Pessimistress Updated with code review!

**Comment by puopg** [comment]
@Pessimistress Ok gonna be totally honest, a lot of this code review woudlve been a lot smoother had I actually built the project and repo. I really just have been editing this single file on github since i thought it was gonna be like a 2 liner.

Super happy you guys have a linter setup for this, I believe I was not assigning userData on the current "deck" variable. I believe it should be `newDeck` for the code in the google maps overlay package.

Anyways, I updated that just now, can you please take another idiot check for me? Sorry for the inconvenience!

**Comment by puopg** [comment]
Going to update with a video, i finally have time today to actually get my development env setup for deck.gl!

**Comment by puopg** [comment]
### Before

https://user-images.githubusercontent.com/7741481/177919178-df278403-1c32-4c81-91a2-3fbbb8e90c24.mp4


### After

https://user-images.githubusercontent.com/7741481/177919145-db17c2a1-4554-4b80-a40a-2f38aabbecf4.mp4

**Comment by Pessimistress** [comment]
CI failure is unrelated to this PR. I will investigate.

**Pull request merged**